packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -148,16 +148,19 @@ class AOTSnapshotter {
     // TODO(cbracken): replace IOSArch with TargetPlatform.ios_{armv7,arm64}.
     assert(platform != TargetPlatform.ios || iosArch != null);
 
-    final bool compileToSharedLibrary = preferSharedLibrary && androidSdk.ndkCompiler != null;
-    if (preferSharedLibrary && !compileToSharedLibrary) {
-      printStatus('Could not find NDK compiler. Not building in shared library mode.');
+    // buildSharedLibrary is Android-only.
+    assert(platform != TargetPlatform.ios || !buildSharedLibrary);
+
+    if (buildSharedLibrary && androidSdk.ndkCompiler == null) {
```

Comment:
do you plan to make `ndkCompiler` easy to configure with an additional option or this would be a part of a separate PR?

I think current error is not actionable - which makes it confusing. Consider morphing it into:

`Could not find NDK compiler at ${ndkPath}. Unable to build with --build-shared-library. Configure ${ndkPath} by setting --ndk-path flag`

or something a long these lines

Reply:
I've updated the error message to be more actionable. For the moment we continue to look for the NDK bundle in the SDK, but we can followup with a flag that allows an override if that's something that would be useful.